JavaScript/JavaScriptSDK/TelemetryContext.ts

```diff
@@ -145,12 +161,35 @@ module Microsoft.ApplicationInsights {
 
             sessionStateEnvelope.time = Util.toISOStringForIE8(new Date(timestamp));
 
-            tc._track(sessionStateEnvelope); 
+            tc._track(sessionStateEnvelope);
+        }
+
+        private _applyCustomProperties(envelope: Microsoft.Telemetry.Envelope) {
+            if (this.properties && envelope && envelope.data && (<any>envelope.data).baseData) {
+                var telemetryItem = (<any>envelope.data).baseData;
+                telemetryItem.properties = telemetryItem.properties || {};
+                Object.getOwnPropertyNames(this.properties).forEach((val, index, array) => {
```

Comment:
Would a [for...in](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/for...in) be simpler here?